Framework/Source/Effects/ToneMapping/ToneMapping.cpp

```diff
@@ -153,64 +160,53 @@ namespace Falcor
-        }
-        else
-        {
-            mpToneMapPass->getProgram()->removeDefine("_ENABLE_EYE_ADAPTATION");
-        }
-        
         //Set shared vars
         mpToneMapVars->getDefaultBlock()->setTexture("gColorTex", pSrc);
         mpLuminanceVars->getDefaultBlock()->setTexture("gColorTex", pSrc);
         mpToneMapVars->getDefaultBlock()->setSampler("gColorSampler", mpPointSampler);
-        mpToneMapVars->getDefaultBlock()->setSampler("gColorSampler", mpLinearSampler);
+        mpLuminanceVars->getDefaultBlock()->setSampler("gColorSampler", mpLinearSampler);
 
         //Calculate luminance
         pRenderContext->setGraphicsVars(mpLuminanceVars);
         pState->setFbo(mpLuminanceFbo);
         mpLuminancePass->execute(pRenderContext);
         mpLuminanceFbo->getColorTexture(0)->generateMips(pRenderContext);
 
+        if (mEnableEyeAdaptation)
+        {
+            mpAdaptationCBuffer["deltaTime"] = deltaTime;
+
+            mpAdaptationVars->getDefaultBlock()->setSampler("gLuminanceTexSampler", mpLinearSampler);
+            mpAdaptationVars->getDefaultBlock()->setTexture("gLuminanceTex", mpLuminanceFbo->getColorTexture(0));
+            mpAdaptationVars->getDefaultBlock()->setTexture("gPreviousFrameTex", mpAdaptationTextures[mAdaptionTexIndex]);
+
+            pRenderContext->setGraphicsVars(mpAdaptationVars);
+            pState->setFbo(mpAdaptationFbo);
+            mpAdaptionPass->execute(pRenderContext);
+            mpAdaptationFbo->getColorTexture(0)->generateMips(pRenderContext);
+        }
+
         //Set Tone map vars
         if (mOperator != Operator::Clamp)
         {
             mpToneMapCBuffer->setBlob(&mConstBufferData, 0u, sizeof(mConstBufferData));
+            mpAdaptationCBuffer->setBlob(&mConstBufferData, 0u, sizeof(mConstBufferData));
             mpToneMapVars->getDefaultBlock()->setSampler("gLuminanceTexSampler", mpLinearSampler);
-            mpToneMapVars->getDefaultBlock()->setTexture("gLuminanceTex", mpLuminanceFbo->getColorTexture(0));
+            mpToneMapVars->getDefaultBlock()->setTexture("gLuminanceTex", mEnableEyeAdaptation ? mpAdaptationFbo->getColorTexture(0) : mpLuminanceFbo->getColorTexture(0));
         }
 
         //Tone map
         pRenderContext->setGraphicsVars(mpToneMapVars);
         pState->setFbo(pDst);
         mpToneMapPass->execute(pRenderContext);
-    }
 
-    float ToneMapping::calculateEV100()
-    {
-        float shutterSpeed = 1.0f;
-        float apertureRadius = 0.0f;
-        
-        if (!mpCamera)
-        {
-            logWarning("No camera set in render pass. Unable to execute eye adaption.");
-        }
-        else
+        if (mEnableEyeAdaptation)
         {
-            shutterSpeed = 1.0f / mpCamera->getFocalLength();
-            apertureRadius = mpCamera->getApertureRadius();
+            mpAdaptationFbo->attachColorTarget(mpAdaptationTextures[mAdaptionTexIndex], 0);
+            mAdaptionTexIndex = !mAdaptionTexIndex;
```

Comment:
I'm surprised the compiler didn't give you a warning on this. Do `1 - mAdaptionIndex` instead